src/server.py

```diff
@@ -0,0 +1,49 @@
+"""Set up a simple HTTP server."""
+
+import socket
+
+
+def server():
+    """Start a server at localhost:5353."""
+    server = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_TCP)
+    address = ('127.0.0.1', 5353)
+    server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    server.bind(address)
+    server.listen(1)
+
+    while True:
+        conn, addr = server.accept()
+        try:
+            total_message = ""
+            buffer_length = 1024
+            while total_message.count("\r\n\r\n") != 2:
+                part = conn.recv(buffer_length)
+                total_message += part.decode('utf8')
+            total_message = total_message.split("\r\n\r\n")
+            msg_body = total_message[1]
+            try:
+                final_message = response_ok() + "".join(msg_body)
+                print(final_message, type(final_message))
+                conn.sendall(final_message.encode('utf8'))
+            except:
+                conn.sendall(response_error().encode('utf8'))
+        except KeyboardInterrupt:
+            break
+    conn.close()
```

Comment:
Similar to your echo server, this line right here is causing your "already in use" errors. You're creating a new `conn` inside your while loop everytime a new request comes in, but the call to close is on the outside of the loop.